### DISCUSSION TITLE
How do I disable non-speaking noise descriptions?

### DISCUSSION THREAD

**Discussion by gth001** [ggml-org/whisper.cpp#735] [Q&A]
I only want speaking to become text, not (buzzing) (clicking) etc...

**Comment by indokid9999** [comment]
Try using sed .
For descriptive sounds that have square brackets try
>sed "s/.\[.*\]/   /" <yourinputfile.

for desciptive sounds that have normal brackets try
>sed "s/.(.*)/   /" <yourinputfile.

For files that have both types of brackets, try
>sed -e "s/.(.*)/   /" -e "s/.\[.*\]/   /" <yourinputfile.

**Reply by aehlke** [reply]
any newer tricks or is this still the go-to?

**Reply by bobqianic** [reply]
Consider using either LLaMA or ChatGPT to eliminate these non-speaking tokens.

**Reply by aehlke** [reply]
I'm now using libfvad in addition to filter what audio is sent to whisper

**Reply by dgoryeo** [reply]
@aehlke , was there any special reason/feature to select libfvad? Many implementations seem to have selected silero-vad, and I was wondering if libfvad might be a more suitable choice for transcription. Would love to hear your thoughts.

**Reply by aehlke** [reply]
I didn't find many options quickly but found how to use it quickly and trusted I could port to web later. I'll check out silero-vad thanks

Oh  yes I saw this... it's much better... I need to change to it

I also see people using gain threshold, like the audio has to be above a floor loudness, in addition. Maybe not necessary with silero.

Another topic - know state of art for speaker rotation? I have found only tinydiarize which the author stopped working on because of their employer

**Reply by dgoryeo** [reply]
I haven't worked with diarization. I tried out few implementations that use pyannote.audio but somehow it doesn't work very well for me --I primarily work with Japanese audio.

**Reply by aehlke** [reply]
May I ask what you're working on? I'm also dealing with Japanese audio, https://reader.manabi.io

**Reply by dgoryeo** [reply]
Games and anime. In those content usually the back ground noise, sounds and music make whisper hallucinate. My current flow is to run 2 transcriptions, one with high vad threashold of 8, and the other with low threshold of 2, then to merge the results. Would love to hear about your experience, tips and tricks.

**Reply by dgoryeo** [reply]
PS. @aehlke , just checked out the `reader.manabi.io`, nice.

**Reply by aehlke** [reply]
I've shared my source here https://github.com/lake-of-fire/SwiftWhisperStream which is for Swift and in a state of flux. Nothing unique yet besides the swift streaming implementation and ios + mac targets, and libfvad which I need to replace.

I'm going to add youtube and mpv features to manabi reader so I want to figure out how to run their audio through whisper to get better transcripts if possible
